docs/user-guide/kubeadm-reference.md

```diff
@@ -0,0 +1,95 @@
+# kubeadm reference
+
+This document specifies the advanced options of kubeadm.
+
+## Usage
+
+### `kubeadm init`
+
+It is usually sufficient to run `kubeadm init` with the default parameters,
+but in some cases you might like to override the default behaviour.
+Here we specify all the flags that can be used to customise the Kubernetes
+installation.
+
+- `--api-advertise-addresses` (multiple values are allowed by having multiple
+flag declarations or multiple values separated by commas)
+- `--api-external-dns-names` (multiple values are allowed by having multiple
+flag declarations or multiple values separated by commas)
+
+By default, `kubeadm` will automatically detect IP addresses and use
+these to generate certificates for the API server. If you would like
+to access the API server through an IP address that can't be automatically
+detected, or through a hostname, you can override these defaults with
+`--api-advertise-addresses` and `--api-external-dns-names`.
+
+- `--cloud-provider`
+
+Currently, `kubeadm` does not provide autodetection of cloud provider.
+This means that networking and persistant volumes are not supported out
+of the box. You can specify a cloud provider using `--cloud-provider`.
+Valid values are the ones support by `controller-manager`, namely `"aws"`,
+`"azure"`, `"cloudstack"`, `"gce"`, `"mesos"`, `"openstack"`, `"ovirt"`,
+`"rackspace"`, `"vsphere"`.
```

Comment:
We need to document what you need to do to make a cloud providers integration work. @errordeveloper?